File: src/main/java/org/blacksun/graph/channel/SimplexChannel.java
package org.blacksun.graph.channel;

import org.blacksun.graph.node.GraphNode;

public class SimplexChannel extends AbstractChannel {
    public SimplexChannel(GraphNode fromNode, GraphNode toNode, int weight, double errors) {
        super(fromNode, toNode, weight, errors);
        connect();
    }

    @Override
    protected String getDirection() {
        return "-->";
    }
}

File: src/main/java/org/blacksun/graph/channel/SimplexChannel.kt
package org.blacksun.graph.channel

import org.blacksun.graph.node.GraphNode

class SimplexChannel(fromNode: GraphNode, toNode: GraphNode, weight: Int, errors: Double) : AbstractChannel(fromNode, toNode, weight, errors) {

    override val direction: String
        get() = "-->"

    init {
        connect()
    }
}
